### PR TITLE
feat(product): split package into more specific subpackages

### PR DESCRIPTION
**Pull request by lderrickable**
BREAKING CHANGE: Many import references for the product package have changed.

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/graycoreio/daffodil/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?
```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## Does this PR introduce a breaking change?
```
[x] Yes
[ ] No
```

### CODE REVIEW

File: libs/product/state/src/resolvers/product-page-server-side/product-page-server-side.resolver.spec.ts [hunk 1]
```diff
@@ -5,9 +5,10 @@ import { ɵPLATFORM_SERVER_ID, ɵPLATFORM_BROWSER_ID } from '@angular/common';
 import { PLATFORM_ID } from '@angular/core';
 import { combineReducers, StoreModule } from '@ngrx/store';
 
+import { daffProductReducers } from '@daffodil/product/state';
+
 import { DaffProductPageServerSideResolver } from './product-page-server-side.resolver';
 import { DaffProductPageResolver } from '../product-page/product-page.resolver';
```
Comment (griest024) [inline libs/product/state/src/resolvers/product-page-server-side/product-page-server-side.resolver.spec.ts:11]: I think `DaffProductPageResolver` should be a `@daffodil` import

File: libs/product/state/src/selectors/configurable-product/configurable-product.selectors.integration.spec.ts [hunk 2]
```diff
@@ -3,13 +3,13 @@ import { StoreModule, combineReducers, Store, select } from '@ngrx/store';
...
 	DaffProductLoadSuccess,
 	daffProductReducers,
 	DaffProductReducersState,
-	DaffConfigurableProductApplyAttribute,
-} from '@daffodil/product';
+	DaffConfigurableProductApplyAttribute
+} from '@daffodil/product/state';
 
 import { getDaffConfigurableProductSelectors } from './configurable-product.selectors';
 import { getDaffConfigurableProductEntitiesSelectors } from '../configurable-product-entities/configurable-product-entities.selectors';
```
Comment (griest024) [inline libs/product/state/src/selectors/configurable-product/configurable-product.selectors.integration.spec.ts:15]: import from `@daffodil/*`

File: libs/product/testing/src/factories/magento/bundle/bundle.factory.ts [hunk 3]
```diff
@@ -3,7 +3,7 @@ import { Injectable } from '@angular/core';
 import {
   DaffModelFactory,
 } from '@daffodil/core/testing';
-import { MagentoProductTypeEnum, MagentoBundledProduct } from '@daffodil/product';
+import { MagentoProductTypeEnum, MagentoBundledProduct } from '@daffodil/product/driver/magento';
```
Comment (griest024) [inline libs/product/testing/src/factories/magento/bundle/bundle.factory.ts:6]: these magento factories should go in `@daffodil/product/driver/magento/testing`

**Review by griest024 — CHANGES_REQUESTED** [review]

File: libs/category/src/category.module.ts [hunk 4]
```diff
@@ -12,7 +13,7 @@ import { DaffProductModule } from '@daffodil/product';
      * Ngrx/store
      */
     DaffCategoryStateModule,
-    DaffProductModule
+    DaffProductStateModule
```
[the remark this answers is not in the document]
Comment (lderrickable) [reply libs/category/src/category.module.ts]: moved

File: libs/product/state/testing/src/public_api.ts [hunk 5]
```diff
@@ -0,0 +1,6 @@
+export { MockDaffCompositeProductFacade } from './mock-composite-product-facade';
+export { MockDaffConfigurableProductFacade } from './mock-configurable-product-facade';
+export { MockDaffProductFacade } from './mock-product-facade';
+export { MockDaffProductGridFacade } from './mock-product-grid-facade';
+export { isProduct } from './product-helper';
```
[the remark this answers is not in the document]
Comment (lderrickable) [reply libs/product/state/testing/src/public_api.ts]: done

File: libs/product/state/src/facades/composite-product/composite-product.facade.spec.ts [hunk 6]
```diff
@@ -6,15 +6,17 @@ import { Dictionary } from '@ngrx/entity';
...
+	DaffCompositeConfigurationItem
+} from '@daffodil/product';
+import {
 	DaffProductLoadSuccess,
 	daffProductReducers,
 	DaffProductReducersState
-} from '@daffodil/product';
+} from '@daffodil/product/state';
 import { DaffCompositeProductFactory } from '@daffodil/product/testing';
+import { DaffCompositeProductApplyOption } from '@daffodil/product/state';
```
[the remark this answers is not in the document]
Comment (lderrickable) [reply libs/product/state/src/facades/composite-product/composite-product.facade.spec.ts]: removed

File: libs/product/state/src/facades/helpers.ts [hunk 7]
```diff
@@ -1,4 +1,4 @@
-import { DaffPriceRange } from '../models/prices';
+import { DaffPriceRange } from '@daffodil/product';
 
 /**
  * Returns whether a DaffPriceRange has a discount.
```
[the remark this answers is not in the document]
Comment (lderrickable) [reply libs/product/state/src/facades/helpers.ts]: moved

File: libs/product/state/src/resolvers/product-page-server-side/product-page-server-side.resolver.spec.ts (hunk 1 above)
Comment (lderrickable) [reply libs/product/state/src/resolvers/product-page-server-side/product-page-server-side.resolver.spec.ts]: done

File: libs/product/state/src/selectors/configurable-product/configurable-product.selectors.integration.spec.ts (hunk 2 above)
Comment (lderrickable) [reply libs/product/state/src/selectors/configurable-product/configurable-product.selectors.integration.spec.ts]: done

File: libs/product/testing/src/factories/magento/bundle/bundle.factory.ts (hunk 3 above)
Comment (lderrickable) [reply libs/product/testing/src/factories/magento/bundle/bundle.factory.ts:6]: moved

**Review by griest024 — DISMISSED** [review]

**Comment by damienwebdev** [comment]
This needs the commits split apart per package, but otherwise LGTM.

**Comment by lderrickable** [comment]
@damienwebdev needs approved

**Review by damienwebdev — APPROVED** [review]
lgtm

**Comment by damienwebdev** [comment]
Merged manually.

**Pull request closed without merging**